src/dayview.html

```diff
@@ -1,187 +1,181 @@
-                          <td class="calendar-hour-column text-center">
-                              {{hourColumnLabels[i]}}
-                          </td>
-                          <td class="calendar-cell" tappable (click)="select(tm.time, tm.events)">
-                              <ng-template [ngTemplateOutlet]="dayviewNormalEventSectionTemplate"
-                                  [ngTemplateOutletContext]="{tm:tm, hourParts: hourParts, eventTemplate:dayviewNormalEventTemplate}">
-                              </ng-template>
-                          </td>
-                      </tr>
-                  </tbody>
-              </table>
-          </init-position-scroll>
-          <init-position-scroll *ngIf="2!==currentViewIndex" class="dayview-normal-event-container"
-              [initPosition]="initScrollPosition">
-              <table class="table table-bordered table-fixed dayview-normal-event-table">
-                  <tbody>
-                      <tr *ngFor="let tm of views[2].rows; let i = index">
-                          <td class="calendar-hour-column text-center">
-                              {{hourColumnLabels[i]}}
-                          </td>
-                          <td class="calendar-cell">
-                              <ng-template [ngTemplateOutlet]="dayviewInactiveNormalEventSectionTemplate"
-                                  [ngTemplateOutletContext]="{tm:tm, hourParts: hourParts}">
-                              </ng-template>
-                          </td>
-                      </tr>
-                  </tbody>
-              </table>
-          </init-position-scroll>
-      </div>
+        </div>
+        <init-position-scroll *ngIf="sliderIndex === currentViewIndex"
+          class="dayview-normal-event-container dayview-categorized-normal-event-container"
+          [initPosition]="initScrollPosition" [emitEvent]="preserveScrollPosition"
+          (onScroll)="setScrollPosition($event)">
+          <table class="table table-bordered table-fixed dayview-normal-event-table">
+            <tbody>
+              <tr *ngFor="let row of views[sliderIndex].rows; let i = index">
+                <td class="calendar-hour-column text-center">
+                  {{ hourColumnLabels[i] }}
+                </td>
+                <td *ngIf="dayviewCategorySource?.size === 0" class="calendar-cell" (click)="select(row)"
+                  tappable>
+                  <ng-template [ngTemplateOutlet]="dayviewNormalEventSectionTemplate"
+                    [ngTemplateOutletContext]="{tm: { events: [], time: row.time }, hourParts, eventTemplate: dayviewNormalEventTemplate}">
+                  </ng-template>
+                </td>
+                <td *ngFor="let category of dayviewCategorySource" class="calendar-cell"
+                  (click)="select(row, category)" tappable>
+                  <ng-template *ngIf="row.eventsGroupByCategory?.get(category)?.length"
```

Comment:
I think even current category doesn't have event, you still want to display the style for dayviewNormalEventSectionTemplate, so should not use *ngIf to exclude it. Otherwise the cells having event and not having event may have different style.